source/extensions/filters/http/jwt_authn/extractor.cc

```diff
@@ -22,6 +22,36 @@ namespace HttpFilters {
 namespace JwtAuthn {
 namespace {
 
+/**
+ * Check Issuer specified in Provider
+ */
+class JwtIssuerChecker {
+public:
+  // add a specified issuer from JwtProvider.
+  void add(const std::string& issuer) {
+    // If a specified issuer is empty, it means to allow all issuers.
+    if (issuer.empty()) {
+      allow_all_ = true;
+    } else {
+      specified_issuers_.insert(issuer);
+    }
+  }
+
+  // check if a jwt issuer is allowed
+  bool check(const std::string& jwt_issuer) const {
+    if (allow_all_) {
+      return true;
+    }
+    return specified_issuers_.find(jwt_issuer) != specified_issuers_.end();
+  }
+
+private:
+  // If true, all issuers are ok
+  bool allow_all_{false};
+  // Only these specified issuers are allowed.
+  absl::node_hash_set<std::string> specified_issuers_;
```

Comment:
Nit: `absl::flat_hash_set` would be a better choice

go/hashmaps#recommendation-2

Reply:
Done